imgui-knobs.cpp

```diff
@@ -9,6 +9,46 @@
 
 namespace ImGuiKnobs {
     namespace detail {
+
+
+
+    //Re-maps a number from one range to another.
+    //Numbers outside the range are not clamped to 0 and 1.
+    template<typename T>
+    inline T Map(const T& aValue,const T& aIStart,const T& aIStop,const T& aOStart,const T& aOStop)
+    {
+      return aOStart + (aOStop - aOStart) * ((aValue - aIStart) / (aIStop - aIStart));
+    }
+
+
+    // Re-Maps clamped
+    template<typename T>
+    inline T MapC(const T& aValue,const T& aIStart,const T& aIStop,const T& aOStart,const T& aOStop)
+    {
+      if(aValue<=aIStart)
+        {
+          return aOStart;
+        }
+      else if(aValue>=aIStop)
+        {
+          return aOStop;
+        }
+      return Map<T>(aValue,aIStart,aIStop,aOStart,aOStop);
+    }
+
+    inline float GetAngle(const ImVec2& aPos,const ImVec2& aMousePos)
+    {
+      ImVec2 dir(aMousePos.x-aPos.x,aMousePos.y-aPos.y);
+      float angle=atan2(-dir.y,-dir.x);
+      angle+=(3.141592f*2.5f);
```

Comment:
Can you use the Pi constant instead? Here and other places

Reply:
Yes, sorry. I was using the previous version with IMGUIKNOBS_PI using double and forgot to fix it when I updated.